src/providers/bitcoin/BitcoinLedgerProvider.js

```diff
@@ -111,15 +110,17 @@ export default class BitcoinLedgerProvider extends LedgerProvider {
     return ((numInputs * 148) + (numOutputs * 34) + 10) * feePerByte
   }
 
-  async getUtxosForAmount (amount, feePerByte = 3, maxAddresses = 20) {
+  async getUtxosForAmount (amount, feePerByte = 3) {
     const utxosToUse = []
     let addressIndex = 0
     let currentAmount = 0
     let numOutputsOffset = 0
 
-    while ((currentAmount < amount) && maxAddresses > 0) {
-      const path = this.getDerivationPathFromIndex(addressIndex)
-      const address = await this.getAddressFromDerivationPath(path)
+    while ((currentAmount < amount)) {
+      const address = await this.getAddressFromIndex(addressIndex)
+      const isAddressUsed = await this.getMethod('isAddressUsed')(address.address)
```

Comment:
My only concern with this approach is that it requires now 2 network calls per loop, while earlier first 20 iteration required 1 network call.

Can we write a hybrid? Scan first 20 and then do this until we find an unused address?

| Addresses to be scanned | New approach | Hybrid approach |
|---|---|---|
| 5 addresses | 10 calls | 20 calls |
| 10 addresses | 20 calls | 20 calls |
| 20 addresses | 40 calls | 20 calls |
| 40 addresses | 80 calls | 60 calls) |

Reply:
implemented that